source/linux/cpu_usage_sampler.c

```diff
@@ -53,33 +61,41 @@ static void aws_get_cpu_sample_fn_linux_get_uint64_delta(uint64_t first, uint64_
     }
 }
 
-static int aws_get_cpu_sample_fn_linux(struct aws_cpu_sampler *sampler, double *output) {
-    struct aws_cpu_sampler_linux *sampler_linux = sampler->impl;
+static int aws_get_cpu_sample_fn_linux(struct aws_system_cpu_sampler *sampler, double *output) {
+    struct aws_system_cpu_sampler_linux *sampler_linux = sampler->impl;
 
     int return_result = AWS_OP_ERR;
-    uint64_t total_user, total_user_low, total_system, total_idle, total;
-    s_get_cpu_usage_linux(&total_user, &total_user_low, &total_system, &total_idle);
-    // total_combined needs to be double to allow for fractions
-    double percent, total_combined;
-
+    // Have to define these here for cleanup
+    uint64_t total_user = 0, total_user_low = 0, total_system = 0, total_idle = 0;
     uint64_t total_user_delta = 0, total_user_low_delta = 0, total_system_delta = 0, total_idle_delta = 0;
+
+    return_result = s_get_cpu_usage_linux(&total_user, &total_user_low, &total_system, &total_idle);
+    if (return_result != AWS_OP_SUCCESS) {
+        *output = 0;
+        goto cleanup;
+    }
+
     aws_get_cpu_sample_fn_linux_get_uint64_delta(total_user, sampler_linux->cpu_last_total_user, &total_user_delta);
     aws_get_cpu_sample_fn_linux_get_uint64_delta(
         total_user_low, sampler_linux->cpu_last_total_user_low, &total_user_low_delta);
     aws_get_cpu_sample_fn_linux_get_uint64_delta(
         total_system, sampler_linux->cpu_last_total_system, &total_system_delta);
     aws_get_cpu_sample_fn_linux_get_uint64_delta(total_idle, sampler_linux->cpu_last_total_idle, &total_idle_delta);
 
-    total_combined = (double)(total_user_delta) + (double)(total_user_low_delta) + (double)(total_system_delta);
-    total = total_combined + (double)(total_idle_delta);
+    double total_combined = (double)(total_user_delta) + (double)(total_user_low_delta) + (double)(total_system_delta);
+    uint64_t total = total_combined + (double)(total_idle_delta);
 
-    if (total == 0) {
-        *output = 0;
-        return_result = AWS_OP_ERR;
-        goto cleanup;
+    double percent = 0;
+    if (total != 0) {
+        percent = (total_combined / total);
     }
 
-    percent = (total_combined / total) * 100;
+    // Clamp to min/max
```

Comment:
You may only need to check for > 1 since immediately following there's a < 0 check that will result in an error